platform/default/src/mbgl/storage/offline_download.cpp

```diff
@@ -387,6 +391,19 @@ void OfflineDownload::deactivateDownload() {
     buffer.clear();
 }
 
+bool OfflineDownload::flushResourcesBuffer() {
+    if (buffer.empty()) return true;
+    try {
+        offlineDatabase.putRegionResources(id, buffer, status);
```

Comment:
should we clear buffer if `putRegionResources` throws `MapboxTileLimitExceededException`?